controllers/storagecluster/reconcile.go

```diff
@@ -448,9 +448,11 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			statusutil.SetErrorCondition(&instance.Status.Conditions, reason, message)
 
 			if instance.Status.Phase != statusutil.PhaseOnboarding {
-				// if the error was due to skipped storage classes
+				// if the error was due to skipped storage classes or
+				// due to StorageProfile list retrieval failure
 				// instead of error phase we will set it to progressing
-				if strings.Contains(returnErr.Error(), storageClassSkippedError) {
+				if strings.Contains(returnErr.Error(), storageClassSkippedError) ||
+					strings.Contains(returnErr.Error(), fmt.Sprint(ocsv1.StorageProfileListError)) {
```

Comment:
I believe this addition to be wrong, the point was to skip only if we get a skip error not every error from the reconciliation. If you want an error to be ignored you will have to return a skip error.

Reply:
This was made to address the missing remediation in a scenario of a failed `StorageCluster` reconciliation due to a failure of some `StorageProfile` or the set of `StorageProfile`s, though in this particular flow apart from the listing error, there aren't any other designated `StorageProfile` errors to check for. Does this equate to _"every error from the reconciliation"_? not sure. It's an added condition to allow re-queuing of the event, meaning nothing but an internal implementation detail, so why does the exact content of its underlying string matter?

Reply:
> This was made to address the missing remediation in a scenario of a failed StorageCluster reconciliation due to a failure of some StorageProfile or the set of StorageProfiles, though in this particular flow apart from the listing error, there aren't any other designated StorageProfile errors to check for.

But in the case of storage profile reconciliation error you need to continue reconciliation for that exact position in the code, not to jump one level up and more to the next ensure. Assuming that you agree with my analysis the correct way to handle it is to just not return in case of an error on the code that reconciles storage profiles. just log the message and move on.